### DISCUSSION TITLE
Stateful apps backups : should databases be stopped first?

### DISCUSSION THREAD

**Discussion by vbartro** [vmware-tanzu/velero#3835] [Community support Q&A]
Hello all,
I could not find a clear answer to my concern : I have a set of mongo and postgreSQL databases, running in k8s, to backup. They use PVs and PVC. I already use restic to back them up. In order to ensure backup data consistency and restore capability, what I currently do  : 
* I either first export data (mongodump and pg_dump) and then back up the export file
*  or I stop the database and back up the content of the database directory from the PV

I'd like to go with Velero, but I'm unclear as to Velero+restic ensures that backups for such stateful, running applications like databases are always consistent.
* Should I continue exporting data, or stop the database first (through prebackup hooks apparently)?
* Or is this now unnecessary, because velero and restic would magically ensure that data will be backed up properly and consistently? 

Thanks

**Comment by jenting** [comment]
You could consider calling the `fsfreeze` before taking the snapshot https://velero.io/docs/v1.6/backup-hooks/#hook-example-with-fsfreeze
